packages/xo-server/src/xapi/index.js

```diff
@@ -272,30 +272,34 @@ export default class Xapi extends XapiBase {
   }
 
   async setHostProperties(id, { nameLabel, nameDescription }) {
-    await this._setObjectProperties(this.getObject(id), {
-      nameLabel,
-      nameDescription,
-    })
+    const host = this.getObject(id)
+    await Promise.all([
+      nameDescription !== undefined &&
+        host.set_name_description(nameDescription),
```

Comment:
`host.set_name_description` doesn't support undefined params ?

Reply:
It does not because it does not make a lot of sense to `host.set_name_description(undefined)`, it's unclear what it does.

IMHO, it's not necessary enough to justify this special case.